parlai/agents/examples/seq2seq.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+Example TorchGeneratorAgent model.
+
+This demonstrates the minimal structure of a building a generative model.
+
+You can train this agent to a reasonable accuracy with:
+
+.. code-block:: bash
+
+    python examples/train_model.py -m examples/seq2seq \
+        -mf /tmp/example_model \
+        -t convai2 -bs 32 -eps 2 --truncate 128
+
+Afterwards, you can play with --beam-size to see how responses differ with
+different beam lengths.
+"""  # noqa: E501
+
+import torch.nn as nn
+import torch.nn.functional as F
+import parlai.core.torch_generator_agent as tga
+
+
+class Encoder(nn.Module):
+    """
+    Example encoder.
```

Comment:
can you elaborate on this a bit more? (and decoder too).. i.e. it instantiates a 1 layer LSTM with hidden size...

Reply:
Yes - added more detail